src/stan/mcmc/hmc/static_uniform/unit_e_static_uniform.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef STAN_MCMC_HMC_STATIC_UNIFORM_UNIT_E_STATIC_UNIFORM_HPP
+#define STAN_MCMC_HMC_STATIC_UNIFORM_UNIT_E_STATIC_UNIFORM_HPP
+
+#include <stan/mcmc/hmc/static_uniform/base_static_uniform.hpp>
+#include <stan/mcmc/hmc/hamiltonians/unit_e_point.hpp>
+#include <stan/mcmc/hmc/hamiltonians/unit_e_metric.hpp>
+#include <stan/mcmc/hmc/integrators/expl_leapfrog.hpp>
+
+namespace stan {
+  namespace mcmc {
+    /**
+      * Hamiltonian Monte Carlo implemetnation that uniformly samples
+      * from trajectories with a static integration time with a
+      * Gaussian-Euclidean disintegration and unit metric
+    */
+    template <typename M, class BaseRNG>
+    class unit_e_static_uniform
+      : public base_static_uniform<M, unit_e_metric, expl_leapfrog, BaseRNG> {
+    public:
+      unit_e_static_uniform(M &m, BaseRNG& rng):
```

Comment:
If M is the model class here, shouldn't it be const?  I don't think a model has anything mutable in it.